portfolio/src/main/webapp/calendarApi.js

```diff
@@ -0,0 +1,298 @@
+// require(['dotenv']).config({path: './.env'});
+
+var GoogleAuth;
+
+// Scopes for API access to Google Calendar
+const SCOPE_READ_ONLY = 'https://www.googleapis.com/auth/calendar.readonly';
+const SCOPE_READ_WRITE = 'https://www.googleapis.com/auth/calendar';
+
+// For the discovery document for Google Calendar API.
+const DISCOVERY_URL =
+    'https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest';
+
+const CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+// Constants for error codes during OAuth process.
+const ERROR_CODE_POPUP_CLOSED = 'popup_closed_by_user';
+const ERROR_CODE_ACCESS_DENIED = 'access_denied';
+
+/**
+ * Loads the API's client and auth2 modules.
+ * Calls the initClient function after the modules load.
+ */
+function initiateCalendarAuth() {
+  gapi.load('client:auth2', initClient);
+}
+
+/** Starts authentication flow based on current user's login status. */
+function initClient() {
+  // Hide the error messages by default.
+  // These messages will appear if authentication errors are caught.
+  hideErrorMessages();
+
+  // Initializes the gapi.client object, which app uses to make API requests.
+  // Initially, the scope is read-only to view user's Google Calendar.
+  gapi.client
+      .init({
+        'apiKey': API_KEY,
+        'clientId': CLIENT_ID,
+        'discoveryDocs': [DISCOVERY_URL],
+        'scope': SCOPE_READ_ONLY
+      })
+      .then(function() {
+        GoogleAuth = gapi.auth2.getAuthInstance();
+
+        // Listen for sign-in state changes.
+        GoogleAuth.isSignedIn.listen(updateCalendarView);
+
+        // Handle initial sign-in state. (Determine if user is already signed
+        // in.)
+        handleAuth();
+      });
+}
+
+/** Signs user in if not logged in, and signs user out otherwise. */
+function handleAuth() {
+  if (GoogleAuth.isSignedIn.get()) {
+    GoogleAuth.signOut();
+  } else {
+    // User is not signed in. Start Google auth flow.
+    GoogleAuth.signIn().catch(function(error) {
+      handleImportAuthError(error);
+    });
+  }
+}
+
+/**
+ * Updates import message box based on the error during authentication process
+ * for importing
+ */
+function handleImportAuthError(e) {
+  $('#import-calendar-message').removeClass('d-none');
```

Comment:
Consider assigning $('...') to a variable.

var $importCalendarMessage = $('#...');

Reply:
Updated.